app/src/main/java/xyz/sangcomz/chameleonsample/ChameleonAdapter.kt

```diff
@@ -5,40 +5,36 @@ import android.view.View
 import android.view.ViewGroup
 import androidx.appcompat.widget.AppCompatImageView
 import androidx.appcompat.widget.AppCompatTextView
+import androidx.recyclerview.widget.RecyclerView
 import com.squareup.picasso.Picasso
 
 /**
  * Created by sangcomz on 27/03/2018.
  */
-class ChameleonAdapter :
-    androidx.recyclerview.widget.RecyclerView.Adapter<ChameleonAdapter.ViewHolder>() {
+class ChameleonAdapter : RecyclerView.Adapter<ChameleonAdapter.ChameleonViewHolder>() {
     private var chameleonList: List<Chameleon> = arrayListOf()
 
     fun setChameleonList(chameleonList: List<Chameleon>) {
         this.chameleonList = chameleonList
         notifyDataSetChanged()
     }
 
-    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
-        return ViewHolder(
-            LayoutInflater.from(parent.context).inflate(R.layout.item_chamelon, parent, false)
-        )
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ChameleonViewHolder {
+        val view: View = LayoutInflater.from(parent.context).inflate(R.layout.item_chamelon, parent, false)
+        return ChameleonViewHolder(view)
     }
 
     override fun getItemCount(): Int = chameleonList.size
 
-    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+    override fun onBindViewHolder(holder: ChameleonViewHolder, position: Int) {
         holder.setItem(chameleonList[position])
     }
 
 
-    class ViewHolder(itemView: View) :
-        androidx.recyclerview.widget.RecyclerView.ViewHolder(itemView) {
+    inner class ChameleonViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
```

Comment:
I know, if there is no code that needs to access the outer class, it doesn't need to use the inner class.
Please explain why you declared it as an inner class.
thank you :)

Reply:
1. As you say, if `inner class` does not use properties in `outer class`, do not need to use `inner class`.

2. I think although multiple adapters can use same `ViewHolder`, but `ViewHolder` is dependent on `Adapter`. also it is not often that use ViewHolder objects alone.

3. One of the feature , `inner class` is not accessible from another class.
* if viewholder is declared inner class
![image](https://user-images.githubusercontent.com/37705123/66255395-e185fa80-e7bd-11e9-8247-9f5aa0f20cac.png)
can not accessiable from onCreate() in MainActivity
![image](https://user-images.githubusercontent.com/37705123/66255406-1abe6a80-e7be-11e9-806e-faa3ab4af7db.png)

4. so. I want to explicitly declare `ViewHolder` as inner class so that this `ViewHolder` is only used on this adapter(outer class).

Reply:
I've never used viewholders as inner classes when developing. This time I learned about `non static inner class` and` static inner class`.

In the current sample app, using it as an inner class seems to be no problem. However, while searching,
>Why do I think so? When the ViewHolder is a non-static inner class of the adapter, it keeps a reference to the adapter. Now, when you call RecyclerView.swapAdapter(newAdapter, false) (or was it true? I don't remember), the new adapter will use the ViewHolders created previously by the old one. As it is impossible to null/clear such an implicit reference in these holders, this reference to the first adapter has leaked and cannot be garbage collected. This is bad enough.

https://stackoverflow.com/questions/29068175/viewholder-not-as-a-inner-class

I think i should find out a little bit more, so if i use like this, i should use `static inner class`.
Thank you :)